### PR TITLE
fix(server): fix compatibility with rdb snapshot

### PR DESCRIPTION
**Pull request by adiholden**
The bug: writing full sync cut opcode to rdb snapshot format breaks compatibility with rdb format

PR changes:
Do not write dfly full sync cut opcode for disk snapshot (not for df and not for rdb)
Add pytest for loading rdb snapshot generated by dragonfly with Redis server

### CODE REVIEW

File: tests/dragonfly/snapshot_test.py [hunk 1]
```diff
@@ -124,6 +126,32 @@ async def test_dbfilenames(
...
+
+    await async_client.connection_pool.disconnect()
+    df_server.stop()
+
+    redis_local_server.start(dir=tmp_dir, dbfilename="test-redis-load-rdb.rdb")
+    await asyncio.sleep(1)
+    c_master = aioredis.Redis(port=redis_local_server.port)
+    await c_master.ping()
+
+    assert await c_master.dbsize() == dbsize
```
Comment (dranikpg) [inline tests/dragonfly/snapshot_test.py:152]: First redis compatibility test 👍🏻 

If more checks are needed, you can re-define `dragonfly.ihash` in redis. It's bit of a hassle, but doable 🙂 

```lua
local dragonfly = {}
function draognfly.ihash(hash, sort, ...)
  local res = redis.pcall(arg)
  -- if sort, sort res
  -- use any hash function to res... lua really doesn't have one :(
end
```

**Review by dranikpg — APPROVED** [review]

File: tests/dragonfly/snapshot_test.py (hunk 1 above)
Comment (adiholden) [reply tests/dragonfly/snapshot_test.py:152]: I wanted to use the seeder capture but got error
redis.exceptions.ResponseError: Error compiling script (new function): user_script:1: unexpected symbol near '#'

I want the fix to go to 1.19, so I will merge this now and can investigate more if we can do a simple change to make this work

Comment (romange) [reply tests/dragonfly/snapshot_test.py:152]: @adiholden  you got an error with Redis, right?

Comment (adiholden) [reply tests/dragonfly/snapshot_test.py:152]: do you mean the script error? yes its in redis

**Review by romange — APPROVED** [review]

**Pull request merged**